.github/docker-demo.yml

```diff
@@ -0,0 +1,21 @@
+name: docker-demo
+
+on:
+  pull_request:
+  push:
+    branches: [main]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Build the stack
+        run: docker-compose up -d
+      - name: Check running containers
+        run: docker ps -a
+      - name: Check logs
+        run: docker logs kglab-notebooks
+      - name: Test all notebooks in the exmaples/ folder can run
+        run: |
+          docker exec -it kglab-notebooks bash -c 'pip install treon && treon work/examples/'
```

Comment:
In order to make sure that people trying the demo notebooks can run all notebooks,
I leverage `treon` https://blog.reviewnb.com/introducing-treon/, a library that will run all notebooks at a given PATH.

In other words:
If any of the notebooks fails in the CI, we know the new commit broke one of the demo notebooks